tests/system/runtimes/test_nuclio.py

```diff
@@ -246,18 +259,114 @@ def kafka_consumer(self):
+        import io
+
+        import avro.schema
+        from avro.io import DatumWriter
+
+        from .map_avro import AVRO_SCHEMA
+
+        for row_index, _ in df.iterrows():
+            event_row_temp = df.loc[[row_index]]
+            event_row_dict = event_row_temp.to_dict("records")[0]
+            writer = DatumWriter(AVRO_SCHEMA)
+            bytes_writer = io.BytesIO()
+            encoder = avro.io.BinaryEncoder(bytes_writer)
+            writer.write(
+                event_row_dict,
+                encoder,
+            )
+            raw_bytes = bytes_writer.getvalue()
+            kafka_producer.send(self.topic_out, raw_bytes)
+            kafka_producer.flush()
+
+    @pytest.mark.skipif(
+        not brokers, reason="MLRUN_SYSTEM_TESTS_KAFKA_BROKERS not defined"
+    )
+    def test_kafka_source_with_avro(self, kafka_fixture):
+
+        row_divide = 3
+        stocks_df = pd.DataFrame(
+            {
+                "ticker": ["MSFT", "GOOG", "AAPL", "CSCO", "AAPL", "AAPL"],
+                "name": [
+                    "Microsoft Corporation",
+                    "Alphabet Inc",
+                    "Apple Inc",
+                    "Cisco Systems Inc",
+                    "Apple Inc",
+                    "Apple Inc",
+                ],
+                "price": [int(30), int(40), int(50), int(20), int(60), int(70)],
+            }
+        )
+        fs_name = "stocks_set"
+        stocks_set = fstore.FeatureSet(fs_name, entities=[fstore.Entity("ticker")])
+
+        # need to set full_event = True since we need to change event md (key) in the Map step
+        stocks_set.graph.to("MyMap", full_event=True)
+        fstore.ingest(
+            featureset=stocks_set,
+            source=stocks_df[0:row_divide],
+            infer_options=fstore.InferOptions.default(),
+        )
+        stocks_set.save()
+
+        kafka_source = KafkaSource(
+            brokers=self.brokers,
+            topics=self.topic_out,
+            initial_offset="earliest",
+            group="my_group",
+        )
+
+        func = mlrun.code_to_function(
+            name="map",
+            kind="serving",
+            # TODO - remove reference to assaf758/mlrun (once mlrun/mlrun is updated with this fix)
+            image="assaf758/mlrun:ML-2836",
+            requirements=["avro"],
+            filename="/home/assafb/iguazio/mlrun/tests/system/runtimes/map_avro.py",
+        )
+
+        run_config = fstore.RunConfig(local=False, function=func).apply(
+            mlrun.auto_mount()
+        )
+        stocks_set_endpoint = fstore.deploy_ingestion_service(
+            featureset=stocks_set,
+            source=kafka_source,
+            run_config=run_config,
+        )
+        print(stocks_set_endpoint)
+
+        kafka_consumer, kafka_producer = kafka_fixture
+        self.produce_kafka_helper(kafka_producer, stocks_df[row_divide:])
+
+        time.sleep(90)  # wait for ingestion-service parquet to be written
```

Comment:
How about we reduce the runtime by setting [flush_after_seconds](https://github.com/mlrun/mlrun/blob/v1.2.0/mlrun/datastore/targets.py#L721)?

Reply:
done